### PR TITLE
Nubis proxy merge

### PR DESCRIPTION
**Pull request by limed**
Merges `nubis-proxy` into `nubis-nat`

### CODE REVIEW

**Comment by limed** [comment]
PR #11

File: nubis/puppet/files/proxy/confd-squid/conf.d/squid_allow_domains.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[template]
+src = "squid_allow_domains.tmpl"
+dest = "/etc/squid/allow_domains.txt"
+prefix = "/%%STACK%%/%%ENVIRONMENT%%"
```
Comment (gozer) [inline nubis/puppet/files/proxy/confd-squid/conf.d/squid_allow_domains.toml]: Now that this is deployed as a sub-stack of the main stack, %%STACK%% in confd will end up being the name of the sub-stack, something really ugly (i.e us-west-2-vpc-StageVPCStack-16CF72VSC8SHF-NATStack-VYX4EPU6OWRL/) and not discoverable. I'd go with %%PROJECT%% instead (i.e. nubis-nat-stage)

File: nubis/puppet/files/proxy/confd-squid/conf.d/squid_allow_regex_domains.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[template]
+src = "squid_allow_regex_domains.tmpl"
+dest = "/etc/squid/allow_regex_domains.txt"
+prefix = "/%%STACK%%/%%ENVIRONMENT%%"
```
Comment (gozer) [inline nubis/puppet/files/proxy/confd-squid/conf.d/squid_allow_regex_domains.toml]: Now that this is deployed as a sub-stack of the main stack, %%STACK%% in confd will end up being the name of the sub-stack, something really ugly (i.e us-west-2-vpc-StageVPCStack-16CF72VSC8SHF-NATStack-VYX4EPU6OWRL/) and not discoverable. I'd go with %%PROJECT%% instead (i.e. nubis-nat-stage)

File: nubis/puppet/files/proxy/confd-squid/conf.d/squid_allow_subnets.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[template]
+src = "squid_allow_subnets.tmpl"
+dest = "/etc/squid/allow_subnets.txt"
+prefix = "/%%STACK%%/%%ENVIRONMENT%%"
```
Comment (gozer) [inline nubis/puppet/files/proxy/confd-squid/conf.d/squid_allow_subnets.toml]: Now that this is deployed as a sub-stack of the main stack, %%STACK%% in confd will end up being the name of the sub-stack, something really ugly (i.e us-west-2-vpc-StageVPCStack-16CF72VSC8SHF-NATStack-VYX4EPU6OWRL/) and not discoverable. I'd go with %%PROJECT%% instead (i.e. nubis-nat-stage)

File: nubis/puppet/files/proxy/confd-squid/conf.d/squid_deny_domains.toml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+[template]
+src = "squid_deny_domains.tmpl"
+dest = "/etc/squid/deny_domains.txt"
+prefix = "/%%STACK%%/%%ENVIRONMENT%%"
```
Comment (gozer) [inline nubis/puppet/files/proxy/confd-squid/conf.d/squid_deny_domains.toml]: Now that this is deployed as a sub-stack of the main stack, %%STACK%% in confd will end up being the name of the sub-stack, something really ugly (i.e us-west-2-vpc-StageVPCStack-16CF72VSC8SHF-NATStack-VYX4EPU6OWRL/) and not discoverable. I'd go with %%PROJECT%% instead (i.e. nubis-nat-stage)

File: nubis/puppet/files/proxy/confd-squid/conf.d/squid_deny_subnets.toml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+[template]
+src = "squid_deny_subnets.tmpl"
+dest = "/etc/squid/deny_subnets.txt"
+prefix = "/%%STACK%%/%%ENVIRONMENT%%"
```
Comment (gozer) [inline nubis/puppet/files/proxy/confd-squid/conf.d/squid_deny_subnets.toml]: Now that this is deployed as a sub-stack of the main stack, %%STACK%% in confd will end up being the name of the sub-stack, something really ugly (i.e us-west-2-vpc-StageVPCStack-16CF72VSC8SHF-NATStack-VYX4EPU6OWRL/) and not discoverable. I'd go with %%PROJECT%% instead (i.e. nubis-nat-stage)

File: nubis/puppet/files/proxy/confd-squid3/conf.d/squid_allow_domains.toml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+[template]
+src = "squid_allow_domains.tmpl"
+dest = "/etc/squid3/allow_domains.txt"
+prefix = "/%%STACK%%/%%ENVIRONMENT%%"
```
Comment (gozer) [inline nubis/puppet/files/proxy/confd-squid3/conf.d/squid_allow_domains.toml]: Now that this is deployed as a sub-stack of the main stack, %%STACK%% in confd will end up being the name of the sub-stack, something really ugly (i.e us-west-2-vpc-StageVPCStack-16CF72VSC8SHF-NATStack-VYX4EPU6OWRL/) and not discoverable. I'd go with %%PROJECT%% instead (i.e. nubis-nat-stage)

File: nubis/puppet/files/proxy/confd-squid3/conf.d/squid_allow_regex_domains.toml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+[template]
+src = "squid_allow_regex_domains.tmpl"
+dest = "/etc/squid3/allow_regex_domains.txt"
+prefix = "/%%STACK%%/%%ENVIRONMENT%%"
```
Comment (gozer) [inline nubis/puppet/files/proxy/confd-squid3/conf.d/squid_allow_regex_domains.toml]: Now that this is deployed as a sub-stack of the main stack, %%STACK%% in confd will end up being the name of the sub-stack, something really ugly (i.e us-west-2-vpc-StageVPCStack-16CF72VSC8SHF-NATStack-VYX4EPU6OWRL/) and not discoverable. I'd go with %%PROJECT%% instead (i.e. nubis-nat-stage)

File: nubis/puppet/files/proxy/confd-squid3/conf.d/squid_deny_domains.toml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+[template]
+src = "squid_deny_domains.tmpl"
+dest = "/etc/squid3/deny_domains.txt"
+prefix = "/%%STACK%%/%%ENVIRONMENT%%"
```
Comment (gozer) [inline nubis/puppet/files/proxy/confd-squid3/conf.d/squid_deny_domains.toml]: Now that this is deployed as a sub-stack of the main stack, %%STACK%% in confd will end up being the name of the sub-stack, something really ugly (i.e us-west-2-vpc-StageVPCStack-16CF72VSC8SHF-NATStack-VYX4EPU6OWRL/) and not discoverable. I'd go with %%PROJECT%% instead (i.e. nubis-nat-stage)

File: nubis/puppet/files/proxy/confd-squid3/conf.d/squid_deny_subnets.toml [hunk 9]
```diff
@@ -0,0 +1,10 @@
+[template]
+src = "squid_deny_subnets.tmpl"
+dest = "/etc/squid3/deny_subnets.txt"
+prefix = "/%%STACK%%/%%ENVIRONMENT%%"
```
Comment (gozer) [inline nubis/puppet/files/proxy/confd-squid3/conf.d/squid_deny_subnets.toml]: Now that this is deployed as a sub-stack of the main stack, %%STACK%% in confd will end up being the name of the sub-stack, something really ugly (i.e us-west-2-vpc-StageVPCStack-16CF72VSC8SHF-NATStack-VYX4EPU6OWRL/) and not discoverable. I'd go with %%PROJECT%% instead (i.e. nubis-nat-stage)

File: README.md [hunk 10]
```diff
@@ -1,27 +1,123 @@
...
 ```
 
-## Configure the deployment
-Create a nubis/cloudformation/parameters.json file by copying the [parameters.json-dist](nubis/cloudformation/parameters.json-dist) file and editing the parameter values. More detailed instructions can be found [here](nubis/cloudformation/README.md#set-up).
+This file is installed by default in [nubis-base](https://github.com/Nubisproject/nubis-base) as */etc/profile.d/proxy.sh*.
+
+## Custom configuration
+To add custom rules to the proxy you need to place some information in consul. The following examples demonstrate how to accomplish this. Note that setting any of these values in consul overwrites the default. Therefore if you wish to include the default you must do so explicatly.
+
+To start with, set up the environment from user data. We actually only need ```$NUBIS_STACK``` and ```$NUBIS_ENVIRONMENT``` which you could set manually.
```
Comment (gozer) [inline README.md]: I'd fix the README as well for the STACK vs PROJECT change

**Pull request merged**